src/components/core/menu/mainMenu/MainMenuRightChainSelector.tsx

```diff
@@ -0,0 +1,74 @@
+import { ReactComponent as IconCheck } from 'assets/icons/check.svg';
+import { FC, useState } from 'react';
+import { DropdownMenu } from 'components/common/dropdownMenu';
+import { buttonStyles } from 'components/common/button/buttonStyles';
+import { carbonEvents } from 'services/events';
+import { cn } from 'utils/helpers';
+
+interface Props {
+  networks: {
+    id: string;
+    name: string;
+    logoUrl: string;
+    isCurrentNetwork: boolean;
+    selectNetwork: () => void;
+  }[];
+}
+
+export const MainMenuRightChainSelector: FC<Props> = ({ networks }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const activeNetwork = networks.find((network) => network.isCurrentNetwork);
+  if (!activeNetwork) return;
+
+  return (
+    <DropdownMenu
+      isOpen={isOpen}
+      setIsOpen={setIsOpen}
+      placement="bottom"
+      className="space-y-2 rounded-[12px] p-8"
+      aria-expanded={isOpen}
+      button={(attr) => (
+        <button
+          {...attr}
+          className={cn(
+            buttonStyles({ variant: 'secondary' }),
+            'relative flex size-40 items-center justify-center p-0'
+          )}
+          onClick={(e) => {
+            setIsOpen(true);
+            carbonEvents.navigation.navNetworkClick(undefined);
+            attr.onClick(e);
+          }}
+        >
+          <img
+            alt={`Select ${activeNetwork.name}`}
+            src={activeNetwork.logoUrl}
+            className="w-20"
+          />
+        </button>
+      )}
+    >
+      {networks.map((network) => {
+        const { id, name, logoUrl, isCurrentNetwork } = network;
+        return (
+          <button
```

Comment:
Use `<a href={href}>` instead of buttons.
You can add `aria-current="page"` & `aria-disabled`